scripts/plugins/check_auto_backup_results.py

```diff
@@ -108,12 +108,26 @@ def main():
     # * the download_path musth be valid
     try:
         backup_results = json.loads(raw_backup_results)
+        # "ERROR" will contain a traceback if something crashed during the backups.
+        # See scripts/templates/auto_backup.py::AutoJujuBackupAll.run()
         if "ERROR" in backup_results:
             msg = "Detected error when performing backup: '{}'".format(
                 backup_results["ERROR"]
             )
             nagios_exit(NAGIOS_STATUS_CRITICAL, msg)
 
+        # This entry is populated by the jujubackupall backup process
+        # (see `BackupTracker.add_error` in jujubackupall),
+        # and indicates which backups failed.
+        # Some backups may have succeeded, but any backup failure
+        # should be considered a critical error,
+        # because silent failed backups
+        # can result in inability to recover from data loss events.
+        if "errors" in backup_results:
```

Comment:
Potentially, but I'd rather keep this new logic separate and clear, especially since there are no tests for this. Maybe we can do some refactoring later - a better result would be to consolidate errors into the same format, so we only need to check one key.

Reply:
The comment do point out there are different scenario. So I am fine with what we have now.

It will be nice to refactor them to separate small function, but it's nice to have.

Reply:
I also think that while we could be more DRY, the gains would be limited: you could reuse the `msg` and the `nagios_exit` call, but you'd still need to extract the separate key value pairs